plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/ptf/Query.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.mongodb.ptf;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.airlift.slice.Slice;
+import io.trino.plugin.mongodb.MongoColumnHandle;
+import io.trino.plugin.mongodb.MongoMetadata;
+import io.trino.plugin.mongodb.MongoSession;
+import io.trino.plugin.mongodb.MongoTableHandle;
+import io.trino.spi.TrinoException;
+import io.trino.spi.connector.ColumnHandle;
+import io.trino.spi.connector.ColumnSchema;
+import io.trino.spi.connector.ConnectorSession;
+import io.trino.spi.connector.ConnectorTableHandle;
+import io.trino.spi.connector.ConnectorTableSchema;
+import io.trino.spi.connector.ConnectorTransactionHandle;
+import io.trino.spi.connector.SchemaTableName;
+import io.trino.spi.ptf.AbstractConnectorTableFunction;
+import io.trino.spi.ptf.Argument;
+import io.trino.spi.ptf.ConnectorTableFunction;
+import io.trino.spi.ptf.ConnectorTableFunctionHandle;
+import io.trino.spi.ptf.Descriptor;
+import io.trino.spi.ptf.ScalarArgument;
+import io.trino.spi.ptf.ScalarArgumentSpecification;
+import io.trino.spi.ptf.TableFunctionAnalysis;
+import org.bson.Document;
+import org.bson.json.JsonParseException;
+
+import javax.inject.Inject;
+import javax.inject.Provider;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static io.trino.spi.StandardErrorCode.INVALID_FUNCTION_ARGUMENT;
+import static io.trino.spi.ptf.ReturnTypeSpecification.GenericTable.GENERIC_TABLE;
+import static io.trino.spi.type.VarcharType.VARCHAR;
+import static java.util.Locale.ENGLISH;
+import static java.util.Objects.requireNonNull;
+
+public class Query
+        implements Provider<ConnectorTableFunction>
+{
+    public static final String SCHEMA_NAME = "system";
+    public static final String NAME = "query";
+
+    private final MongoMetadata metadata;
+
+    @Inject
+    public Query(MongoSession session)
+    {
+        requireNonNull(session, "session is null");
+        this.metadata = new MongoMetadata(session);
+    }
+
+    @Override
+    public ConnectorTableFunction get()
+    {
+        return new RawQueryFunction(metadata);
+    }
+
+    public static class RawQueryFunction
```

Comment:
Should it be `QueryFunction` or `RawQueryFunction` ?

Reply:
Renamed to `QueryFunctionHandle` likes `trino-base-jdbc` module.

Reply:
bump ?

Reply:
Should the inner class be re-named as `QueryFunction` ?

Reply:
Sorry, I renamed another `Raw` usage. Now:
*  QueryFunction extends AbstractConnectorTableFunction
*  QueryFunctionHandle implements ConnectorTableFunctionHandle